File: tests/resources/test_codegen/test_default_code_generators_and_templates/nim/echo_template.nim
import sequtils
proc scanf(formatstr: cstring){.header: "<stdio.h>", varargs.}
proc getchar(): char {.header: "<stdio.h>", varargs.}
proc nextInt(): int = scanf("%lld",addr result)
proc nextFloat(): float = scanf("%lf",addr result)
proc nextString(): string =
  var get = false
  result = ""
  while true:
    var c = getchar()
    if int(c) > int(' '):
      get = true
      result.add(c)
    else:
      if get: break
      get = false

{% if mod %}
let MOD = {{ mod }}
{% endif %}
{% if yes_str %}
let YES = "{{ yes_str }}"
{% endif %}
{% if no_str %}
let NO = "{{ no_str }}"
{% endif %}

proc solve({{ formal_arguments }}):void =
  echo N," ",M
  assert H.len == N - 1
  for i in 0..<N-1:
    assert H[i].len == M - 2
    for j in 0..<M-2:
      stdout.write if j > 0: " " else:"", H[i][j]
    echo ""
  assert A.len == N - 1
  assert B.len == N - 1
  for i in 0..<N-1:
    echo A[i]," ",B[i]
  echo Q
  assert X.len == M + Q
  for i in 0..<M + Q:
    echo X[i]
  echo YES
  echo NO
  echo MOD

proc main():void =
  {{input_part}}
  solve({{ actual_arguments }})
  return

main()

File: tests/resources/test_codegen/test_default_code_generators_and_templates/swift/echo_template.swift
import Foundation

{% if mod is not none %}
let MOD = {{ mod }}
{% endif %}
{% if yes_str is not none %}
let YES = "{{ yes_str }}"
{% endif %}
{% if no_str is not none %}
let NO = "{{ no_str }}"
{% endif %}

func solve({{ formal_arguments }}) -> Bool {
    print("\(N) \(M)")
    guard H.count == N - 1 else {
        return false
    }
    for i in 0..<N-1 {
        guard H[i].count == M - 2 else {
            return false
        }
        for j in 0..<M-2 {
            if j > 0 {
                print(" ", terminator: "")
            }
            print("\(H[i][j])", terminator: "")
        }
        print("")
    }
    guard A.count == N - 1, B.count == N - 1 else {
        return false
    }
    for i in 0..<N-1 {
        print("\(A[i]) \(B[i])")
    }
    print("\(Q)")
    guard X.count == M + Q else {
        return false
    }
    for i in 0..<M + Q {
        print("\(X[i])")
    }

    print(YES)
    print(NO)
    print(MOD)

    return true
}

func main() {
    var tokenIndex = 0, tokenBuffer = [String]()
    func readString() -> String {
        if tokenIndex >= tokenBuffer.count {
            tokenIndex = 0
            tokenBuffer = readLine()!.split(separator: " ").map { String($0) }
        }
        defer { tokenIndex += 1 }
        return tokenBuffer[tokenIndex]
    }
    func readInt() -> Int { Int(readString())! }
    func readDouble() -> Double { Double(readString())! }
    {{input_part}}
    _ = solve({{ actual_arguments }})
}

main()
